src/containers/App/App.js

```diff
@@ -197,281 +161,299 @@ export /* istanbul ignore next */ class App extends Component {
-                    />
-                    <Route
-                      path={paths.triggerTemplates.byNamespace()}
-                      exact
-                      component={TriggerTemplates}
-                    />
-                    <Route
-                      path={paths.extensions.all()}
-                      exact
-                      component={Extensions}
-                    />
-                    {extensions
-                      .filter(extension => !extension.type)
-                      .map(({ displayName, name, source }) => (
-                        <Route
-                          key={name}
-                          path={paths.extensions.byName({ name })}
-                          render={({ match }) => (
-                            <Extension
-                              displayName={displayName}
-                              match={match}
-                              source={source}
-                            />
-                          )}
-                        />
-                      ))}
-
-                    <Route
-                      path={paths.kubernetesResources.all()}
-                      exact
-                      component={ResourceList}
-                    />
-                    <Route
-                      path={paths.kubernetesResources.byNamespace()}
-                      exact
-                      component={ResourceList}
-                    />
-                    <Route
-                      path={paths.kubernetesResources.byName()}
-                      exact
-                      component={CustomResourceDefinition}
-                    />
-                    <Route
-                      path={paths.kubernetesResources.cluster()}
-                      exact
-                      component={CustomResourceDefinition}
-                    />
-                    <Route
-                      path={paths.rawCRD.byNamespace()}
-                      exact
-                      component={CustomResourceDefinition}
-                    />
-                    <Route
-                      path={paths.rawCRD.cluster()}
-                      exact
-                      component={CustomResourceDefinition}
-                    />
-
-                    <Redirect to={urls.pipelineRuns.all()} />
-                  </Switch>
-                </PageErrorBoundary>
-              </Content>
-            </>
-          </Router>
-        )}
-      </IntlProvider>
-    );
+  async function fetchConfig() {
+    setIsLoadingConfig(true);
+    try {
+      await fetchInstallProperties();
+      await loadMessages();
+      setIsLoadingConfig(false);
+      setShowLoadingState(false);
+    } catch (error) {
+      console.error(error); // eslint-disable-line no-console
+      setLoadingConfigError(error);
+      setIsLoadingConfig(false);
+      setShowLoadingState(false);
+    }
   }
+
+  useEffect(() => {
+    if (webSocketConnected !== false) {
```

Comment:
```
if (webSocketConnected) {
...
```

Reply:
By default `useEffect` is called after every render (including the first one after the component is mounted). We're limiting that here by passing `webSocketConnected` in runs after the first render and then only after subsequent renders where the value of `webSocketConnected` has changed.

[`webSocketConnected` is initially `null`](https://github.com/tektoncd/dashboard/blob/4d23fc20367beb5e4370ca89f28dbef0486d0d12/src/reducers/notifications.js#L15) until we attempt to connect. We want to call `fetchConfig` when the `App` component mounts, regardless of the websocket connection status, and later only on reconnect.

As part of the switch to `react-query` we'll also be ditching these status booleans and manual checks, letting react-query handle most of the reconnect / retry logic for us.